src/api/cvc4cppkind.h

```diff
@@ -1832,11 +1832,23 @@ enum CVC4_PUBLIC Kind : int32_t
    */
   MEMBER,
   /**
-   * The set of the single element given as a parameter.
+   * From the API, use Term Solver::mkSingleton(Sort s, Term t)
+   * The SINGLETON operator here should not be used in the API.
+   * Using mkTerm(SINGLETON, Term child) would throw a typing error.
+   * It is not removed here because it is neede by the parser.
+   *
+   * Operator for singleton.
    * Parameters: 1
-   *   -[1]: Single element
+   *   -[1]: The type of the single element
    * Create with:
-   *   mkTerm(Kind kind, Term child)
+   *   NodeManager::mkConst(SingletonOp(t))
+   *
+   * Construct a singleton set from an element given as a parameter.
+   * Parameters: 2
+   *   -[1]: SingletonOp
+   *   -[2]: Single element
+   * Create with:
+   *   mkNode(kind::SINGLETON, op, n);
```

Comment:
The user of the API doesn't need to know about these details of SingletonOp.  This documentation should reference the solver's `mkSingleton(n)` and simply warn not to use `mkTerm(SINGLETON, n)`.

Reply:
Then the user does not need to know about the SINGLETON operator. We can write don't use this kind, but then the user would wonder why is it defined here if it is not needed. I tried to delete it, but the parsers for smt, cvc, tptop complain

Reply:
It is needed, the user will want to know the kind of singleton Terms.